src/plugin.cpp

```diff
@@ -0,0 +1,202 @@
+}
+
+int slurm_spank_init_post_opt(spank_t sp, int ac, char **av) {
+  return impl::slurm_spank_init_post_opt(sp, ac, av);
+}
+
+} // extern "C"
+
+//
+// Implementation
+//
+namespace impl {
+struct arg_pack {
+  std::string uenv_arg;
+  bool uenv_flag_present = false;
+  bool run_prologue = false;
+};
+
+static arg_pack args{};
+
+/// wrapper for spank_getenv
+std::optional<std::string> getenv(spank_t sp, const char *var) {
+  const int len = 1024;
+  char buf[len];
+
+  if (spank_context() == spank_context_t::S_CTX_LOCAL ||
+      spank_context() == spank_context_t::S_CTX_ALLOCATOR) {
+    auto ret = std::getenv(var);
+    if (ret == nullptr) {
+      return std::nullopt;
+    }
+    return ret;
+  }
+
+  spank_err_t ret = spank_getenv(sp, var, buf, len);
+
+  if (ret == ESPANK_ENV_NOEXIST) {
+    return std::nullopt;
+  }
+
+  if (ret == ESPANK_SUCCESS) {
+    return std::string{buf};
+  }
+
+  slurm_spank_log("getenv failed");
+  throw ret;
+}
+
+static spank_option uenv_arg{
+    (char *)"uenv",
+    (char *)"<file>[:mount-point][,<file:mount-point>]*",
+    (char *)"A comma seprated list of file and mountpoint, default mount-point "
+            "is " DEFAULT_MOUNT_POINT,
+    1, // requires an argument
+    0, // plugin specific value to pass to the callback (unnused)
+    [](int val, const char *optarg, int remote) -> int {
+      slurm_verbose("uenv: val:%d optarg:%s remote:%d", val, optarg, remote);
+      args.uenv_flag_present = true;
+      args.uenv_arg = optarg;
+      return ESPANK_SUCCESS;
+    }};
+
+static spank_option prolog_arg{
+    (char *)"uenv-skip-prologue",
+    (char *)"none",
+    (char *)"ignore any environment prologue script",
+    0, // takes an argument
+    0, // plugin specific value to pass to the callback (unnused)
+    [](int val, const char *optarg, int remote) -> int {
+      slurm_verbose("uenv-mount: val:%d optarg:%s remote:%d", val, optarg, remote);
+      args.run_prologue = false;
+      return ESPANK_SUCCESS;
+    }};
+
+/// detect if srun/sbatch has been called from an squashfs-run/squashfs-mount
+std::optional<std::string> get_uenv_env(spank_t sp) {
+  return getenv(sp, UENV_MOUNT_LIST);
+}
+
+int slurm_spank_init(spank_t sp, int ac, char **av) {
+
+  for (auto arg : {&uenv_arg, &prolog_arg}) {
+    if (auto status = spank_option_register(sp, arg)) {
+      return status;
+    }
+  }
+
+  return ESPANK_SUCCESS;
+}
+
+/// check if image, mountpoint is valid
+int init_post_opt_remote(spank_t sp,
+                         const std::vector<mount_entry> &mount_entries) {
+  return do_mount(sp, mount_entries);
+}
+
+/// check if image/mountpoint are valid
+int init_post_opt_local_allocator(
+    spank_t sp, const std::vector<mount_entry> &mount_entries) {
+  for (auto &entry : mount_entries) {
+    switch (entry.p) {
+    case protocol::file: {
+      if (!is_valid_image_and_mountpoint(entry.mount_point, entry.image_path)) {
+        slurm_error("Inexistent image or invalid mountpoint given: %s:%s",
```

Comment:
The validity check in local/allocator context now iterates over all paths separately, print a message in case it is not a valid file/directory and return with error at the end.